lib/adapter.js

```diff
@@ -6873,6 +7012,13 @@ function Adapter(options) {
                 callback && callback('States database not connected');
                 return;
             }
+            if (id === null || id === undefined || id === '') {
+                callback && callback('delBinaryState id missing');
+                return;
+            }
+            if (typeof id !== 'string') {
+                return callback(`ID can be only string and not "${typeof id}"`);
+            }
```

Comment:
This entire block is repeated everywhere. We should refactor it out into its own method:
```js
/**
 * Checks if a passed ID is valid. Returns `true` if it is, `false` otherwise.
 */
function validateId(id, callback) {
   if (id === null || id === undefined || id === '') {
        callback && callback('delBinaryState id missing');
        return false;
    }
    if (typeof id !== 'string') {
        callback && callback(`ID can be only string and not "${typeof id}"`);
        return false;
    }
    return true;
}
```
and call it in every place where it is currently copied:
```js
if (!validateId(id, callback)) return;
```

Reply:
like ... will change later today ...